src/test/scala/io/iohk/ethereum/vm/OpCodeFunSpec.scala

```diff
@@ -701,7 +701,25 @@ class OpCodeFunSpec extends FunSuite with OpCodeTesting with Matchers with Prope
         val (refundDW, stack1) = stateIn.stack.pop
         val world1 = stateIn.world
           .transfer(stateIn.ownAddress, Address(refundDW), stateIn.ownBalance)
-          .saveAccount(Address(refundDW), Account.empty())
+        val expectedState = stateIn
+          .withWorld(world1)
+          .withAddressToDelete(stateIn.context.env.ownerAddr)
+          .withStack(stack1)
+          .halt
+        stateOut shouldEqual expectedState
+      }
+    }
+
+    // Refunding gas to self should destroy all account ether
+    val stateGenSelf = getProgramStateGen(
+      stackGen = getStackGen(elems = 2, uint256Gen = getUInt256Gen(ownerAddr.toUInt256, ownerAddr.toUInt256))
+    )
+    forAll(stateGenSelf) { stateIn =>
```

Comment:
No need to run 10 additional tests (that's what `forAll` with a generator will do) to test this edge case. Anyway this test wasn't great from the start. See the branch `fix/selfdestructUnitTest`, and like last time, merge if you like it.